cmd/medius/images/promote.go

```diff
@@ -0,0 +1,110 @@
+package images
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/repository"
+)
+
+func NewPromoteImagesCommand(options *common.Options) *cobra.Command {
+	options.PromoteImageOptions = common.PromoteImageOptions{
+		TargetRegistry: "quay.io/containerdisks",
+	}
+
+	promoteCmd := &cobra.Command{
+		Use:   "promote",
+		Short: "Promote verified containerdisks from one registry to another registry",
+		Run: func(cmd *cobra.Command, args []string) {
+			results, err := readResultsFile(options.ImagesOptions.ResultsFile)
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			resultsChan, err := spawnWorkers(cmd.Context(), options, func(a api.Artifact) (*api.ArtifactResult, error) {
+				r, ok := results[a.Metadata().Describe()]
+				if !ok {
+					return nil, nil
+				}
+				if r.Err != "" {
+					return nil, fmt.Errorf("Artifact %s failed in stage %s: %s", a.Metadata().Describe(), r.Stage, r.Err)
```

Comment:
Hm, thinking about it, this should return the passed in data, so that we still have a complete file, right?